src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -1009,6 +1009,53 @@ public function testMirrorCopiesRelativeLinkedContents()
         $this->assertEquals('\\' === DIRECTORY_SEPARATOR ? realpath($sourcePath.'\nested') : 'nested', readlink($targetPath.DIRECTORY_SEPARATOR.'link1'));
     }
 
+    public function testMirrorContentsWithSameNameAsSourceOrTargetWithoutDeleteOption()
+    {
+        $sourcePath = $this->workspace.DIRECTORY_SEPARATOR.'source'.DIRECTORY_SEPARATOR;
+
+        $oldPath = getcwd();
```

Comment:
> Doing clean up work after performing assertions is not reliable.

True, I've moved them to directly before and after the call to `mirror()`.

> So what we could is...

Another PR perhaps?